### PR TITLE
clean audio service shutdown

### PR DESCRIPTION
**Pull request by JarbasAl**
## Description
Adds a shutdown method to AudioBackend, this may be required by some backends that interact with other programs

Also cleanly removes listeners

## How to test
Currently no audio backend needs this, in the mpv backend PR #1489  this is necessary and correctly performs shutdown, without this audio continues playing

## Contributor license agreement signed?
CLA [ yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/audio/main.py [hunk 1]
```diff
@@ -449,6 +449,28 @@ def setup_pulseaudio_handlers(self, pulse_choice=None):
                 self.pulse_quiet = self.pulse_lower_volume
                 self.pulse_restore = self.pulse_restore_volume
 
+    def shutdown(self):
+        for s in self.service:
+            LOG.info('shutting down ' + s.name)
+            s.shutdown()
```
Comment (forslund) [inline mycroft/audio/main.py]: Add a try-except block here so a failure in one backend won't cause others to not shutdown.

**Comment by JarbasAl** [comment]
changes made, should i also add a call to the stop method before shutdown? or do i assume the shutdown method always implements it like it should?

**Comment by forslund** [comment]
I think assuming the backend handles stop is the best way. Stop shouldn't be forced by the system here

**Comment by JarbasAl** [comment]
maybe add a call to stop in the backend template class in case some backend doesn't implement this method?

**Comment by forslund** [comment]
That could be a good idea. I guess the times you don't want to do stop is when working against a "remote" backend like the chromecast or mopidy and in these cases they can implement a shutdown that don't call stop.

So go ahead and add that to the service class!

**Comment by forslund** [comment]
Rebased and is now merging!

**Pull request merged**